### ISSUE TITLE
copy/download of published files is broken.

### ISSUE THREAD

**Issue by haraldschilly** [sagemathinc/cocalc#2932]
i.e. consider this [project listing](https://cocalc.com/projects/18c77389-a5c3-49de-946a-7593b53d3fb2/?session=) and sign in as a user who isn't a collaborator or adminstrator.

1. having to sign in isn't always clear, I was able to trigger an error (instead of triggering some warning when trying to copy)
2. selecting several files and trying to download them doesn't work (it can't create a zip file, obviously)
3. the copy operation seems to be outright broken. There is a report about the copy button not being enabled after selecting the destination.

Maybe we should retire that entire listing and somehow redirect to the share server? I'm not sure how to do that, because all the listed files are separate shares.

**Comment by williamstein** [comment]
No we should not retire this.  This is what the share server redirects to
in fact.  The share server obviously doesn't have download and copy to your
project functionality really.   We simply need to fix these bugs.... By
fixing them.

**Comment by williamstein** [comment]
> Maybe we should retire that entire listing and somehow redirect to the share server? I'm not sure how to do that, because all the listed files are separate shares.

OK, yes we should.  Right now, it's a dangerous can of worms, and provides a dangerous way for any random user to start a project running.  I now agree with you on this.

**Comment by williamstein** [comment]
Note that actually this functionality is already completely disabled, for other reasons (related to security).   We need a new **much better** "Open in Cocalc" workflow.

**Comment by williamstein** [comment]
OK, I just tried and as of today NONE of the problems you report above (from a few weeks ago) is a problem now.  Isn't that weird?   I think all these problems may have been a side effect of some websocket api change, which I fixed last week.   In particular:

> 1. having to sign in isn't always clear

If you are not signed in then you are 100% required to sign in before doing anything further now. That seems very clear.

> x2.  selecting several files and trying to download them doesn't work (it can't create a zip file, obviously)

Selecting files doesn't provide any options at all, so there is no Download button to not work.   I think the right fix is to remove the checkboxes... though clicking several, then clicking the blue "Copy" button does work as expected, which is useful.  

> x3. the copy operation seems to be outright broken. 

It worked fine for me.

> There is a report about the copy button not being enabled after selecting the destination.

There's no copy button to enable.
